api/src/repositories/adminRepository.ts

```diff
@@ -0,0 +1,24 @@
+import * as db from 'zapatos/db';
+import { conditions as dc } from 'zapatos/db';
+import type * as s from 'zapatos/schema';
+import { role_type } from 'zapatos/schema';
+
+import pool from '../util/pool';
+
+type userRoleSQL = s.users.SQL | s.user_roles.SQL;
+export type userRoleSelectable = s.users.JSONSelectable & { user_role: role_type };
+
+/**
+ * Gets all users and their roles.
+ * @returns List of all users and their assigned roles.
+ */
+const getUsersInfo = async (): Promise<userRoleSelectable[]> => {
```

Comment:
Definitely need a double check on this (whether or not I'm doing the join correctly)

Reply:
Seems right to me. Is it working and giving you the results you expect?

Reply:
Actually don't you need the role in the SELECT clause?